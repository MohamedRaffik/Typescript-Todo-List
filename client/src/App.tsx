import * as React from 'react';
import * as ReactRouterDOM from 'react-router-dom';
import * as ListComponent from './components/TodoList';
import * as styles from './stylesheets/App.scss';

const { BrowserRouter, Switch, Link, Route } = ReactRouterDOM;
const { TodoList } = ListComponent;

export const App = () => {
	const [Lists, setLists] = React.useState({
		Main: [],
		Test: []
	});
	const [listPages, changeListPages] = React.useState({ current: 1, total: 1 });
	const [itemPages, changeItemPages] = React.useState({ current: 1, total: 1 });

	const sidebarLists = Object.keys(Lists)
		.slice(listPages.current * 25 - 25, listPages.current * 25)
		.map(listName => (
			<Link key={listName} to={`/${listName === 'Main' ? '' : listName}`}>
				{listName}
			</Link>
		));

	const ListRoutes = Object.keys(Lists).map(listName => (
		<Route key={listName} exact={true} path={`/${listName === 'Main' ? '' : listName}`}>
			<TodoList listName={listName} />
		</Route>
	));

	return (
		<BrowserRouter>
			<div className={styles['nav']}>
				<div className={styles['nav-container']}>
					<p className={styles['nav-container-header']}>TodoLists</p>
					<div className={styles['nav-container-account']}>
						<p>Login</p>
						<p>Register</p>
					</div>
				</div>
			</div>
			<div className={styles['pagegrid']}>
				<div className={styles['pagegrid-sidebar']}>
					<div className={styles['pagegrid-sidebar-list']}>{sidebarLists}</div>
					<div className={styles['pagegrid-sidebar-buttongrid']}>
						<button>{'<<'}</button>
						<button>{'<'}</button>
						<button style={{ fontWeight: 'bold' }}>{listPages.current}</button>
						<button>{'>'}</button>
						<button>{'>>'}</button>
					</div>
				</div>
				{/*PlaceHolder*/}
				<div className={styles['pagegrid-mainview']}>
					<Switch>{ListRoutes}</Switch>
				</div>
			</div>
		</BrowserRouter>
	);
};
