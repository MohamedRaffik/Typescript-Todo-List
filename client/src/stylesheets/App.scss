@import './partials.scss';

body {
	@extend .reset-spacing;
	font-family: Arial, Helvetica, sans-serif;
}

.nav {
	background-color: secondary-color(0, 0, 0);
	color: black;

	&-container {
		display: flex;
		padding: 1em 0.5em;

		&-header {
			@extend .reset-spacing;
			font-size: 1.5em;
			flex-grow: 7;
		}

		&-account {
			display: flex;
			flex-grow: 1;
			font-size: 1.1em;

			& > * {
				margin: auto 0.5em;
			}
		}
	}
}

.pagegrid {
	display: flex;

	&-sidebar {
		@extend .reset-spacing;
		flex-grow: 1;
		border: secondary-color(0, 0, 0) solid 0.1em;
		border-top: none;
		border-radius: 0px 0px 5px;

		&-list > a {
			@extend .reset-spacing;
			display: block;
			padding: 1em;
			color: secondary-color(-125, -25, 25);
			text-decoration: none;
		}

		&-buttongrid {
			@extend .reset-spacing;
			display: flex;
			padding: 0.5em 0em;
			justify-content: center;
			background-color: secondary-color(0, 0, 0);

			& > button {
				color: secondary-color(-125, -55, 25);
				background-color: secondary-color(0, 0, 0);
				border: none;
			}
		}
	}

	&-mainview {
		flex-grow: 10;
		padding: 0em 1em;
	}
}
