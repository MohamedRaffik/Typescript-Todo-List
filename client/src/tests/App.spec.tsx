import * as React from 'react';
import * as renderer from 'react-test-renderer';
import * as AppComponent from '../App';

const { App } = AppComponent;

describe('Testing App Component', () => {
	it('should render', () => {
		const component = renderer.create(<App />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
