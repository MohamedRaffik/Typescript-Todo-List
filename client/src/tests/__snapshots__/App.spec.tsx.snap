// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component should render 1`] = `
Array [
  <div>
    <div>
      <p>
        TodoLists
      </p>
      <div>
        <p>
          Login
        </p>
        <p>
          Register
        </p>
      </div>
    </div>
  </div>,
  <div>
    <div>
      <div>
        <a
          href="/"
          onClick={[Function]}
        >
          Main
        </a>
        <a
          href="/Test"
          onClick={[Function]}
        >
          Test
        </a>
      </div>
      <div>
        <button>
          &lt;&lt;
        </button>
        <button>
          &lt;
        </button>
        <button
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          1
        </button>
        <button>
          &gt;
        </button>
        <button>
          &gt;&gt;
        </button>
      </div>
    </div>
    <div>
      <div>
        Main
      </div>
    </div>
  </div>,
]
`;
